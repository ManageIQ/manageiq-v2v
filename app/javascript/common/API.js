import axios from 'axios';

const getAuthToken = () =>
  localStorage.miq_token ? localStorage.miq_token : '';

axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
axios.defaults.headers.common['X-Auth-Token'] = getAuthToken();

export default {
  get(url, headers = {}, params = {}) {
    return axios.get(url, {
      headers,
      params
    });
  },
  put(url, data = {}, headers = {}) {
    return axios.put(url, data, {
      headers
    });
  },
  post(url, data = {}, headers = {}) {
    return axios.post(url, data, {
      headers
    });
  },
  delete(url, headers = {}) {
    return axios.delete(url, {
      headers
    });
  },
  patch(url, data = {}, headers = {}) {
    return axios.patch(url, data, {
      headers
    });
  }
};

export const globalMockMode = true;
