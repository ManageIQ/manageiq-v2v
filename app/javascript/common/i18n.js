import { addLocaleData } from 'react-intl';

// hard wired to english for now
import en from 'react-intl/locale-data/en';
import fr from 'react-intl/locale-data/fr';
import es from 'react-intl/locale-data/es';
import ja from 'react-intl/locale-data/ja';
import zh_CN from 'react-intl/locale-data/zh';
import pt_BR from 'react-intl/locale-data/pt';

class I18n {
  constructor(locale, timezone) {
    zh_CN[0].locale = 'zh_CN';
    pt_BR[0].locale = 'pt_BR';
    this.fallbackIntl = !global.Intl;

    [this.locale] = locale.split('-');
    this.timezone = this.fallbackIntl ? 'UTC' : timezone;

    // this.loaded = false;
    // this.ready = this.init();

    // hard wired for now
    addLocaleData([...en, ...fr, ...es, ...ja, ...zh_CN, ...pt_BR]);
    this.loaded = true;
  }

  // todo: gather react-intl locale-data from the server asynchronously
  // async init() {
  //   await this.fetchIntl();
  //   addLocaleData(await import(`react-intl/locale-data/${this.locale}`));
  //   this.loaded = true;
  // }

  // async fetchIntl() {
  //   if (this.fallbackIntl) {
  //     global.Intl = await import('intl');
  //     await import(`intl/locale-data/jsonp/${this.locale}`);
  //   }
  // }
}

const [htmlElemnt] = document.getElementsByTagName('html');
const langAttr = htmlElemnt.getAttribute('lang') || 'en';
const timezoneAttr = htmlElemnt.getAttribute('data-timezone') || 'UTC';

const i18n = new I18n(langAttr, timezoneAttr);

export default i18n;
