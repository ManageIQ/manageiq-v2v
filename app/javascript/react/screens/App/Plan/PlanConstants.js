export const FETCH_V2V_PLAN_REQUEST = 'FETCH_V2V_PLAN_REQUEST';
export const FETCH_V2V_PLAN = 'FETCH_V2V_PLAN';
export const QUERY_V2V_PLAN_VMS = 'QUERY_V2V_PLAN_VMS';
export const RESET_PLAN_STATE = 'RESET_PLAN_STATE';
