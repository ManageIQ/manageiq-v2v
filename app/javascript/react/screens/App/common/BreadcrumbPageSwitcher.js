import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { OverlayTrigger, Popover, Button, Icon, ListGroup, ListGroupItem } from 'patternfly-react';

const BreadcrumbPageSwitcher = ({ activeHref }) => {
  const SwitcherItem = ({ name, href }) => (
    <ListGroupItem className={classNames('no-border', { active: href === activeHref })} href={href}>
      {name}
    </ListGroupItem>
  );
  SwitcherItem.propTypes = {
    name: PropTypes.string,
    href: PropTypes.string
  };

  // TODO: remove this when we are ready to release analytics
  const hideMigrationAnalytics = true;

  const popoverContent = (
    <ListGroup style={{ marginBottom: 0, borderTop: 0 }}>
      <SwitcherItem name={__('Migration Plans')} href="#/plans" />
      <SwitcherItem name={__('Infrastructure Mappings')} href="#/mappings" />
      <SwitcherItem name={__('Migration Settings')} href="#/settings" />
      {!hideMigrationAnalytics && <SwitcherItem name={__('Migration Analytics')} href="#/analytics" />}
    </ListGroup>
  );

  return (
    <OverlayTrigger
      rootClose
      trigger="click"
      placement="bottom"
      overlay={<Popover id="breadcrumb-switcher-popover">{popoverContent}</Popover>}
    >
      <Button
        style={{
          marginTop: '-2px',
          marginBottom: '-4px',
          marginLeft: '15px',
          height: '22px',
          width: '32px',
          lineHeight: '14px'
        }}
      >
        <Icon type="fa" name="exchange" />
      </Button>
    </OverlayTrigger>
  );
};

BreadcrumbPageSwitcher.propTypes = {
  activeHref: PropTypes.string
};

export default BreadcrumbPageSwitcher;
