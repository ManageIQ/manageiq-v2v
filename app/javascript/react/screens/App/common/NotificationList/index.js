import { connect } from 'react-redux';
import NotificationList from './NotificationList';
import * as NotificationListActions from './NotificationListActions';
import reducer from './NotificationListReducer';

export const reducers = { notifications: reducer };

const mapStateToProps = ({ notifications }, ownProps) => ({
  ...notifications,
  ...ownProps.data
});

const mergeProps = (stateProps, dispatchProps, ownProps) =>
  Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(mapStateToProps, NotificationListActions, mergeProps)(
  NotificationList
);
