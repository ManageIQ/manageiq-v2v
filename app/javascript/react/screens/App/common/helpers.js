import React from 'react';

// TODO remove these, they are space fillers
const t = str => (
  <div align="center">
    <h1>TODO: {str}!</h1>
  </div>
);

export const todo = str => (
  <div>
    {t(str)}
    {t(str)}
    {t(str)}
    {t(str)}
    {t(str)}
  </div>
);
