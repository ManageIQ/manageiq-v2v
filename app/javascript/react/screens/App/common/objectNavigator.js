export default (path, object) =>
  path.split('.').reduce((previous, current) => (previous ? previous[current] : null), object);
