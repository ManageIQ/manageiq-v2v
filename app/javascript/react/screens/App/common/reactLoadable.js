import React from 'react';
import PropTypes from 'prop-types';
import Loadable from 'react-loadable';

const LoadingPage = ({ error, pastDelay, timedOut }) => {
  if (error) {
    return <div>Error! Please refresh and try again.</div>;
  } else if (timedOut) {
    return <div>Timed out. Please refresh and try again.</div>;
  } else if (pastDelay) {
    return <div>Loading...</div>;
  }
  return null;
};

export default options =>
  Loadable({
    ...options,
    loading: LoadingPage,
    delay: 200,
    timeout: 10000
  });

LoadingPage.propTypes = {
  error: PropTypes.bool,
  pastDelay: PropTypes.bool,
  timedOut: PropTypes.bool
};
