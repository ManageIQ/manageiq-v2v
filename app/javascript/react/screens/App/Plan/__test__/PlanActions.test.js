import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { mockRequest, mockReset } from '../../../../../common/mockRequests';

import {
  _getPlanActionCreator,
  fetchPlanAction,
  resetPlanStateAction
} from '../PlanActions';
import { requestPlanData } from '../plan.fixtures';
import { FETCH_V2V_PLAN, RESET_PLAN_STATE } from '../PlanConstants';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('FETCH_V2V_PLAN', () => {
  const id = '1';
  const { fetchPlanUrl } = requestPlanData(id);
  const request = {
    fetchPlanUrl,
    status: 200
  };

  describe('_getPlanActionCreator', () => {
    test('sets up the FETCH_V2V_PLAN action object', () => {
      const dispatch = jest.fn();
      mockRequest(request);

      _getPlanActionCreator(fetchPlanUrl, id)(dispatch);

      expect(dispatch).toHaveBeenLastCalledWith({
        type: FETCH_V2V_PLAN,
        payload: expect.any(Promise)
      });
    });
  });

  describe('fetchPlanAction', () => {
    let urlBuilder;
    beforeEach(() => {
      urlBuilder = jest.fn();
      urlBuilder.mockReturnValueOnce(fetchPlanUrl);
    });

    test('dispatches the PENDING and FULFILLED actions', () => {
      mockRequest(request);
      return store.dispatch(fetchPlanAction(urlBuilder, id)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    test('dispatches the PENDING and REJECTED actions', () => {
      mockRequest({
        ...request,
        status: 404
      });
      return store.dispatch(fetchPlanAction(urlBuilder, id)).catch(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });
});

describe('resetPlanStateAction', () => {
  test('sets up the RESET_PLAN_STATE action object', () => {
    const action = resetPlanStateAction();

    expect(action).toEqual({
      type: RESET_PLAN_STATE
    });
  });
});
