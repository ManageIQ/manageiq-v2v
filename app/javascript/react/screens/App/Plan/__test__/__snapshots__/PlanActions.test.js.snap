// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FETCH_V2V_PLAN fetchPlanAction dispatches the PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_PLAN_PENDING",
  },
  Object {
    "payload": Object {
      "config": Object {
        "data": undefined,
        "headers": Object {
          "Accept": "application/json, text/plain, */*",
          "X-Auth-Token": "",
          "X-Requested-With": "XMLHttpRequest",
        },
        "maxContentLength": -1,
        "method": "get",
        "params": Object {},
        "timeout": 0,
        "transformRequest": Object {
          "0": [Function],
        },
        "transformResponse": Object {
          "0": [Function],
        },
        "url": "/api/dummyPlan",
        "validateStatus": [Function],
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
      },
      "data": null,
      "headers": undefined,
      "status": 200,
    },
    "type": "FETCH_V2V_PLAN_FULFILLED",
  },
]
`;

exports[`FETCH_V2V_PLAN fetchPlanAction dispatches the PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_PLAN_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: Request failed with status code 404],
    "type": "FETCH_V2V_PLAN_REJECTED",
  },
]
`;
